### PR TITLE
update-etcd-rewrite

### PR DESCRIPTION
**Pull request by calvix**
rewrite of https://github.com/giantswarm/k8scloudconfig/pull/193

### CODE REVIEW

**Comment by r7vme** [comment]
k8s does compaction on it's own https://github.com/kubernetes/kubernetes/pull/24079 Why we need additional one?

Copy paste here from other ticket:
Few more things to think about compaction happens every 5 min and gyus say that they have problems with load peaks it causes [1]. So in community they implemented retention modes. [2]

1 - https://github.com/coreos/etcd/issues/8098
2 - https://github.com/coreos/etcd/issues/8123

WDYT @calvix @teemow ? imo we should be careful with such changes. We saw the problem only once and on very specific environment (that survived multiple updates and had inoptimized cronjobs that produced thousands of resources). And finally we are not sure that that it lack of compaction was a root cause. 

In my opinion we should add a monitoing on that first.

**Comment by calvix** [comment]
compaction + defrag was the thing i needed to do in order to restore `lycan` so from my point of view it is the cause of cluster being `f*cked`

I can easily see customer doing same thing on k8s cluster as we did on `lycan`.

**Comment by calvix** [comment]
SO this happened again on Vodafone guest cluster, so it's not something random or specific to  `lycan`

**Comment by r7vme** [comment]
Okay, but i'm still not 100% confident :D

https://github.com/coreos/etcd/issues/8009

TL;DR despite they have autocompaction enabled, they still hit this issue. And it's related to BoltDB bug. Fix  will be landed only in [3.3](https://github.com/coreos/etcd/issues/8009#issuecomment-324393895)

For me it''s still not clear what was the root cause for both cases.

**Comment by corest** [comment]
IMHO, a single appearance of that issue looks like a corner case, which does not bring enough information for applying particular resolving strategy

**Comment by teemow** [comment]
It happened on Lycan and Viking (in a guest cluster). I think this can always happen and we need to monitor that at least. For now starting with the host clusters.

**Comment by calvix** [comment]
https://github.com/giantswarm/giantswarm/issues/1841
